### ISSUE TITLE
does the implementation of biggan as the same efficient as the author's implementation?

### ISSUE THREAD

**Issue by lihuiknight** [POSTECH-CVLab/PyTorch-StudioGAN#58]
Dear author, thanks for your implementation of BigGAN. Does your implementation as the same efficient as [the author's implementation](The author's officially unofficial PyTorch BigGAN implementation)? 
How long does your implementation to achieve the similar IS and FID for ILSVRC2012  on the 8xV100?

**Comment by MINGUKKANG** [comment]
Hi,

In BigGAN-PyTorch repo, the authors noticed that "On 8xV100 with full-precision training (no Tensor cores), this script takes 15 days to train to 150k iterations".

Our repository, however, takes a month (30days) to train a BigGAN using a different training setup to 200k iterations. 

Specifically, 

BigGAN-PyTorch
1. D-step: 1
2. No sync_bn
3. Training iterations: 150K
4. No flip augmentation

Ours
1. D-step: 2
2. Sync_bn
3. Training iterations: 200K
4. Flip augmentation

I think that overall training time is the same in the case of ImageNet training.

Best,

Minguk

**Comment by lihuiknight** [comment]
that's great. does the gradient of generator or discriminator will be crashed during the whole training procedure within 30 days?

**Comment by MINGUKKANG** [comment]
Although BigGAN paper has reported training collapse during training, We do not have experienced the training collapse during BigGAN2056 training. 

I think that it was just a fluke.

Thank you.

**Comment by lihuiknight** [comment]
Cool. It's a really high-quality implementation.  Do you plan to add the latest GAN variants into this repository?

**Comment by MINGUKKANG** [comment]
I will add SSGAN, McGAN, StyleGAN2, and useful regularizations for research. 

But I couldn't decide when to do it. It spends a lot of time to implement and verify the effectiveness of modules. 

Best, 

Minguk

**Comment by lihuiknight** [comment]
Great! Your work is very helpful.
